Index: RcovParser.java
===================================================================
--- RcovParser.java	(revision 22953)
+++ RcovParser.java	(working copy)
@@ -13,6 +13,7 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import org.apache.commons.lang.StringUtils;
 import org.htmlparser.Node;
 import org.htmlparser.Parser;
 import org.htmlparser.Text;
@@ -52,7 +53,7 @@
         	
         	if (report.getRowCount() > 0) {        		
         		  //row at 0 is the header row, so we have to get the row at 1 
-        		  TableRow totalRow = report.getRow(1);        		
+        		  TableRow totalRow = report.getRow(1);
         		  TableColumn[] columns = totalRow.getColumns();
 
               result.setTotalLines(getTextFromTT(columns[1]));
@@ -135,7 +136,9 @@
             Node parent = first.getChildren() != null && first.getChildren().size() > 0?first:first.getParent();
 
             parent.collectInto(nodeList, new NodeClassFilter(Text.class));
-            text = nodeList.elementAt(0).getText();
+            for (int i = 0; i < nodeList.size() && StringUtils.isEmpty(text); i++) {
+                text = nodeList.elementAt(i).getText().trim();
+            }
         }
 
         return text;
