Index: src/main/java/hudson/plugins/twitter/TwitterPublisher.java
===================================================================
--- src/main/java/hudson/plugins/twitter/TwitterPublisher.java	(revision 30679)
+++ src/main/java/hudson/plugins/twitter/TwitterPublisher.java	(working copy)
@@ -147,7 +147,7 @@
                 tinyUrl = "?";
             }
         }
-        return String.format("%s%s:%s $%d - %s", toblame, result, projectName, build.number, tinyUrl);
+        return String.format("%s%s:%s #%d - %s", toblame, result, projectName, build.number, tinyUrl);
     }
 
     private String getUserString(AbstractBuild<?, ?> build) throws IOException {
@@ -316,4 +316,4 @@
             });
         }
     }
-}
\ No newline at end of file
+}
