Index: src/main/resources/hudson/model/RadiatorView/job.jelly
===================================================================
--- src/main/resources/hudson/model/RadiatorView/job.jelly	(revision 39752)
+++ src/main/resources/hudson/model/RadiatorView/job.jelly	(working copy)
@@ -8,7 +8,7 @@
 	<j:if test="${job.building || job.queued}">
 		<j:set var="clazz" value="building" />
 	</j:if>
-	<div style="${jobStyle}  background-color:${job.backgroundColor};"
+	<div style="${jobStyle}  background-color:${job.backgroundColor}; background-image: none;"
 		class="job ${job.getStatus()} ${clazz}" tooltip="${job.name}" id="job">
 
 		<!--  Job Name Title -->
