Index: ResultReader.java
===================================================================
--- ResultReader.java	(revision 38031)
+++ ResultReader.java	(working copy)
@@ -72,7 +72,7 @@
    }
 
    private Test readTest(String testLine, boolean isTotals, boolean hasTPS) {
-      Scanner scanner = new Scanner(testLine).useDelimiter("\\s{2,}").useLocale(Locale.ENGLISH);
+      Scanner scanner = new Scanner(testLine).useDelimiter("\\s{2,}");
       String id = scanner.next();
       int testCount = scanner.nextInt();
       int errorCount = scanner.nextInt();
