diff --git a/core/src/main/java/hudson/tasks/test/TestResultAggregator.java b/core/src/main/java/hudson/tasks/test/TestResultAggregator.java
index 22d90b9..03a2ecb 100644
--- a/core/src/main/java/hudson/tasks/test/TestResultAggregator.java
+++ b/core/src/main/java/hudson/tasks/test/TestResultAggregator.java
@@ -26,6 +26,7 @@ package hudson.tasks.test;
 import hudson.Launcher;
 import hudson.matrix.MatrixAggregator;
 import hudson.matrix.MatrixBuild;
+import hudson.matrix.MatrixConfiguration;
 import hudson.matrix.MatrixRun;
 import hudson.model.BuildListener;
 
@@ -57,4 +58,15 @@ public class TestResultAggregator extends MatrixAggregator {
         if(atr!=null)   result.add(atr);
         return true;
     }
+
+   @Override
+   public boolean endBuild() throws InterruptedException, IOException {
+      for (MatrixConfiguration mc : build.getProject().getActiveConfigurations()) {
+         if (mc.getLastBuild().getNumber() != build.getNumber()) {
+            AbstractTestResultAction atr = mc.getLastCompletedBuild().getAction(AbstractTestResultAction.class);
+            if (atr != null) result.add(atr);
+         }
+      }
+      return true;
+   }
 }
