Index: src/main/java/hudson/plugins/perforce/PerforceSCM.java
===================================================================
--- src/main/java/hudson/plugins/perforce/PerforceSCM.java	(revision 23885)
+++ src/main/java/hudson/plugins/perforce/PerforceSCM.java	(working copy)
@@ -62,6 +62,8 @@
     String p4Port;
     String p4Client;
     String projectPath;
+    /* This is better for build than original options noallwrite noclobber nocompress unlocked nomodtime normdir */
+    String projectOptions = "noallwrite clobber compress unlocked nomodtime rmdir";
     String p4Label;
 
     String p4Exe = "C:\\Program Files\\Perforce\\p4.exe";
@@ -97,7 +99,7 @@
     int firstChange = -1;
 
     @DataBoundConstructor
-    public PerforceSCM(String p4User, String p4Passwd, String p4Client, String p4Port, String projectPath,
+    public PerforceSCM(String p4User, String p4Passwd, String p4Client, String p4Port, String projectPath, String projectOptions,
                        String p4Exe, String p4SysRoot, String p4SysDrive, String p4Label, boolean forceSync,
                        boolean updateView, boolean dontRenameClient, int firstChange, PerforceRepositoryBrowser browser) {
 
@@ -105,6 +107,7 @@
         this.setP4Passwd(p4Passwd);
         this.p4Client = p4Client;
         this.p4Port = p4Port;
+        this.projectOptions = projectOptions;
 
         //make it backwards compatible with the old way of specifying a label
         Matcher m = Pattern.compile("(@\\S+)\\s*").matcher(projectPath);
@@ -655,6 +658,9 @@
 
         p4workspace.setName(p4Client);
 
+        // Set the workspace options according to the configuration
+        p4workspace.setOptions(getProjectOptions());
+
         // Ensure that the root is appropriate (it might be wrong if the user
         // created it, or if we previously built on another node).
 
@@ -1016,6 +1022,22 @@
     }
 
     /**
+     * The current perforce option set for the view.
+     * @return current perforce view options
+     */
+    public String getProjectOptions() {
+        return projectOptions;
+    }
+
+    /**
+     * Set the perforce options for view creation.
+     * @param projectOptions the effective perforce options.
+     */
+    public void setProjectOptions(String projectOptions) {
+        this.projectOptions = projectOptions;
+    }
+
+    /**
      * @param update    True to let the plugin manage the view, false to let the user manage it
      */
     public void setUpdateView(boolean update) {
Index: src/main/java/com/perforce/api/Client.java
===================================================================
--- src/main/java/com/perforce/api/Client.java	(revision 23885)
+++ src/main/java/com/perforce/api/Client.java	(working copy)
@@ -306,6 +306,8 @@
 			p.println("Description: " + getDescription());
 			p.println("View:");
 			p.println(getView());
+                        p.println("Options:");
+			p.println(getOptions());
 			p.flush();
 			p.outClose();
 			while(null != (l = p.readLine())) {
@@ -367,7 +369,7 @@
 							break;
 						addView(l);
 					}
-				}
+				} 
 			}
 			p.close();
 			inSync();
Index: src/main/resources/hudson/plugins/perforce/PerforceSCM/config.jelly
===================================================================
--- src/main/resources/hudson/plugins/perforce/PerforceSCM/config.jelly	(revision 23885)
+++ src/main/resources/hudson/plugins/perforce/PerforceSCM/config.jelly	(working copy)
@@ -40,6 +40,11 @@
         <f:textarea field="projectPath"/>
       </f:entry>
 
+      <f:entry title="Options">
+        <f:textbox field="projectOptions" default="noallwrite clobber compress unlocked nomodtime rmdir"
+             value="${instance.projectOptions}" />
+      </f:entry>
+      
       <f:entry title="One Time Force Sync">
         <f:checkbox field="forceSync" value="True"/>
       </f:entry>
Index: src/main/webapp/help/project.html
===================================================================
--- src/main/webapp/help/project.html	(revision 23885)
+++ src/main/webapp/help/project.html	(working copy)
@@ -24,7 +24,6 @@
     way.  This behavior can be disabled by the option, "Let Hudson Manage Workspace View.  This will
     allow you to retrieve more than one depot into the workspace of a given job by providing your own
     list of views.
-    </table>
   </p>
   <p>
     <b>Let Hudson Manage Workspace View</b>: For most cases, leave this checked.  Hudson will update
@@ -45,6 +44,13 @@
     //depot/folder1/... //workspace-name/folder1/...<br>
     //depot/folder2/... //workspace-name/folder2/...<br>
   </p>
+  <p><b>Options</b><br/>
+      Original setting is <i>noallwrite noclobber nocompress unlocked nomodtime normdir.</i><br>
+      Default is <i>noallwrite clobber compress unlocked nomodtime rmdir.</i><br>
+      The main differences between original setup: clobber files will owerride, empty directories will be remove, and
+      data transfer will be compressed.
+      <a href="http://www.perforce.com/perforce/doc.current/manuals/cmdref/client.html#1040665">See p4 guide.</a>
+  </p>
   <p>
     <b>One Time Force Sync</b>: allows you to tell perforce to do a p4 sync -f //... which will make sure to
     pull down ALL files to the workspace, regardless if they have been sync'd before.  Use this to get around
