diff --git a/src/main/java/hudson/scm/IntegritySCM.java b/src/main/java/hudson/scm/IntegritySCM.java
index e0ae993..49e3ec4 100644
--- a/src/main/java/hudson/scm/IntegritySCM.java
+++ b/src/main/java/hudson/scm/IntegritySCM.java
@@ -565,6 +565,7 @@ public class IntegritySCM extends SCM implements Serializable
 		Command siViewProjectCmd = new Command(Command.SI, "viewproject");
 		siViewProjectCmd.addOption(new Option("recurse"));
 		siViewProjectCmd.addOption(new Option("project", siProject.getConfigurationPath()));
+		siViewProjectCmd.addOption(new Option("filter", "!pending:add"));
 		MultiValue mvFields = new MultiValue(",");
 		mvFields.add("name");
 		mvFields.add("context");
