Index: Functions.java
===================================================================
--- Functions.java	(revision 8382)
+++ Functions.java	(working copy)
@@ -744,8 +744,8 @@
             switch (j.getIconColor().noAnime()) {
             case ABORTED:
             case RED:
+                return "Failure";
             case YELLOW:
-                return "Failure";
             case BLUE:
                 return "Success";
             case DISABLED:
