diff --git a/src/main/java/org/jenkinsci/plugins/p4/client/ClientHelper.java b/src/main/java/org/jenkinsci/plugins/p4/client/ClientHelper.java
index 54f58a85..7f13e3f5 100644
--- a/src/main/java/org/jenkinsci/plugins/p4/client/ClientHelper.java
+++ b/src/main/java/org/jenkinsci/plugins/p4/client/ClientHelper.java
@@ -793,7 +793,7 @@ public class ClientHelper extends ConnectionHelper {
 		}
 
 		List<IFileSpec> status = iclient.reconcileFiles(files, statusOpts);
-		getValidate().check(status, "- no file(s) to reconcile", "instead of", "empty, assuming text", "also opened by");
+		getValidate().check(status, "No file(s) to reconcile", "instead of", "empty, assuming text", "also opened by");
 	}
 
 	public String publishChange(Publish publish) throws Exception {
