Index: src/main/java/hudson/plugins/m2extrasteps/M2ExtraStepsWrapper.java
===================================================================
--- src/main/java/hudson/plugins/m2extrasteps/M2ExtraStepsWrapper.java	(revision 23701)
+++ src/main/java/hudson/plugins/m2extrasteps/M2ExtraStepsWrapper.java	(working copy)
@@ -7,7 +7,9 @@
 import hudson.model.AbstractProject;
 import hudson.model.BuildListener;
 import hudson.model.Descriptor;
+import hudson.model.Hudson;
 import hudson.model.Result;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
 import hudson.tasks.BuildStep;
 import hudson.tasks.BuildWrapper;
 import hudson.tasks.BuildWrapperDescriptor;
@@ -16,6 +18,7 @@
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import net.sf.json.JSONObject;
 
@@ -185,4 +188,11 @@
         }
     }
     
+    public class Environment extends BuildWrapper.Environment {
+        public void buildEnvVars(Map<String, String> env) {      
+          EnvironmentVariablesNodeProperty prop;
+          prop = Hudson.getInstance().getGlobalNodeProperties().get(EnvironmentVariablesNodeProperty.class);      
+          env.putAll(prop.getEnvVars());        	
+        }
+    }    
 }
