diff --git a/src/main/java/net/praqma/jenkins/memorymap/parser/AbstractMemoryMapParser.java b/src/main/java/net/praqma/jenkins/memorymap/parser/AbstractMemoryMapParser.java
index 440c173..bdf7559 100644
--- a/src/main/java/net/praqma/jenkins/memorymap/parser/AbstractMemoryMapParser.java
+++ b/src/main/java/net/praqma/jenkins/memorymap/parser/AbstractMemoryMapParser.java
@@ -55,6 +55,7 @@ import org.apache.commons.collections.ListUtils;
 public abstract class AbstractMemoryMapParser implements Describable<AbstractMemoryMapParser>, ExtensionPoint, MemoryMapParsable, Serializable {
 
     private static final String UTF_8_CHARSET = "UTF8";
+    private static final String ASCII_CHARSET = "Cp1252";
     protected static final Logger logger = Logger.getLogger(AbstractMemoryMapParser.class.toString());
     
     protected List<Pattern> patterns;
@@ -77,7 +78,12 @@ public abstract class AbstractMemoryMapParser implements Describable<AbstractMem
     }
      
     protected CharSequence createCharSequenceFromFile(File f) throws IOException {
-        return createCharSequenceFromFile(UTF_8_CHARSET, f);
+        CharBuffer cbuf = null;
+        try {
+            return createCharSequenceFromFile(UTF_8_CHARSET, f);
+        } catch (IOException ex) {
+            return createCharSequenceFromFile(ASCII_CHARSET, f);
+        }
     }    
      
     protected CharSequence createCharSequenceFromFile(String charset, File f) throws IOException {
