Index: src/main/java/hudson/plugins/bazaar/BazaarChangeLogParser.java
===============================================================
--- src/main/java/hudson/plugins/bazaar/BazaarChangeLogParser.java	(revision 30369)
+++ src/main/java/hudson/plugins/bazaar/BazaarChangeLogParser.java	(working copy)
@@ -36,6 +36,7 @@ public class BazaarChangeLogParser extends ChangeLogParser {
             }
 
             s = s.trim();
+            len = s.length();
             if ("------------------------------------------------------------".equals(s)) {
                 if (entry != null && state > 2) {
                     if (message.length() != 0) {
Index: src/main/java/hudson/plugins/bazaar/BazaarSCM.java
===============================================================
--- src/main/java/hudson/plugins/bazaar/BazaarSCM.java	(revision 30369)
+++ src/main/java/hudson/plugins/bazaar/BazaarSCM.java	(working copy)
@@ -270,7 +270,7 @@ public class BazaarSCM extends SCM implements Serializable {
     }
 
     @Override
-    public void buildEnvVars(AbstractBuild build, Map<String, String> env) {
+    public void buildEnvVars(AbstractBuild<?,?> build, Map<String, String> env) {
     }
 
     @Override
