--- CVSSCM.java	2009/08/12 19:29:00
+++ CVSSCM.java	2009/08/12 19:30:21
@@ -864,7 +864,7 @@
                     task.setFailOnError(true);
                     BufferedOutputStream bufferedOutput = new BufferedOutputStream(out);
                     task.setDeststream(bufferedOutput);
-                    task.setBranch(branch);
+                    task.setTag(isTag() ? ":" + branch : branch);
                     task.setStart(startTime);
                     task.setEnd(endTime);
                     if(changedFiles!=null) {
