diff --git a/src/main/java/hudson/plugins/deploy/CargoContainerAdapter.java b/src/main/java/hudson/plugins/deploy/CargoContainerAdapter.java
index 3a8caaa..d60ab48 100644
--- a/src/main/java/hudson/plugins/deploy/CargoContainerAdapter.java
+++ b/src/main/java/hudson/plugins/deploy/CargoContainerAdapter.java
@@ -23,6 +23,7 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.Serializable;
+import org.codehaus.cargo.container.deployable.EAR;
 
 /**
  * Provides container-specific glue code.
@@ -50,18 +51,22 @@
         configure(config);
         return containerFactory.createContainer(id, ContainerType.REMOTE, config);
     }
-    
+
     protected void deploy( DeployerFactory deployerFactory, final BuildListener listener, Container container, File f, String contextPath ) {
         Deployer deployer = deployerFactory.createDeployer(container);
 
         listener.getLogger().println("Deploying "+f+" to container "+container.getName());
 
         deployer.setLogger(new LoggerImpl(listener.getLogger()));
-        WAR war = createWAR( f );
-        if ( !StringUtils.isEmpty(contextPath)) {
-            war.setContext( contextPath );
+        if (f.getName().endsWith(".war")) {
+            WAR war = createWAR(f);
+            if (!StringUtils.isEmpty(contextPath)) {
+                war.setContext(contextPath);
+            }
+        } else {
+            EAR ear = createEAR(f);
+            deployer.redeploy(ear);
         }
-        deployer.redeploy(war);
     }
 
     /**
@@ -71,6 +76,10 @@
      */
     protected WAR createWAR(File deployableFile) {
         return new WAR(deployableFile.getAbsolutePath());
+    }
+
+    protected EAR createEAR(File deployableFile) {
+        return new EAR(deployableFile.getAbsolutePath());
     }
 
     public boolean redeploy(FilePath war, final String contextPath, AbstractBuild<?, ?> build, Launcher launcher, final BuildListener listener) throws IOException, InterruptedException {
@@ -86,7 +95,7 @@
                 final DeployerFactory deployerFactory = new DefaultDeployerFactory(cl);
 
                 Container container = getContainer(configFactory, containerFactory, getContainerId());
-                
+
                 deploy(deployerFactory, listener, container, f, contextPath);
                 return true;
             }
