Index: maven-plugin/src/main/java/hudson/maven/MavenBuild.java
===================================================================
--- maven-plugin/src/main/java/hudson/maven/MavenBuild.java	(revision 18723)
+++ maven-plugin/src/main/java/hudson/maven/MavenBuild.java	(working copy)
@@ -39,6 +39,7 @@
 import hudson.remoting.Channel;
 import hudson.scm.ChangeLogSet;
 import hudson.scm.ChangeLogSet.Entry;
+import hudson.tasks.BuildTrigger;
 import hudson.tasks.BuildWrapper;
 import hudson.util.ArgumentListBuilder;
 import org.apache.maven.BuildFailureException;
@@ -521,7 +522,11 @@
         }
 
         public void cleanUp(BuildListener listener) throws Exception {
-            if(getResult().isBetterOrEqualTo(Result.SUCCESS))
+            Result threshold = Result.SUCCESS;
+            BuildTrigger trigger = project.getPublishersList().get(BuildTrigger.class);
+            if (trigger != null)
+                threshold = trigger.getThreshold();
+            if(!getResult().isWorseThan(threshold))
                 scheduleDownstreamBuilds(listener,new HashSet<AbstractProject>());
         }
     }
