diff -Nuar docker-workflow-plugin-docker-workflow-1.23.org/src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java docker-workflow-plugin-docker-workflow-1.23/src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java
--- docker-workflow-plugin-docker-workflow-1.23.org/src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java	2020-07-30 11:53:02.288167129 +0200
+++ docker-workflow-plugin-docker-workflow-1.23/src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java	2020-07-30 11:56:31.301345689 +0200
@@ -109,9 +109,9 @@
         argb.add("run", "-t", "-d");
 
         // Username might be empty because we are running on Windows
-        if (StringUtils.isNotEmpty(user)) {
+        /* if (StringUtils.isNotEmpty(user)) {
             argb.add("-u", user);
-        }
+        }*/
         if (args != null) {
             argb.addTokenized(args);
         }
