--- MavenBuilder-1.336.java	2010-01-12 08:55:57.936652400 +0100
+++ MavenBuilder-1.336-patched.java	2010-01-12 08:56:20.086119200 +0100
@@ -217,7 +217,17 @@
     private String formatArgs(List<String> args) {
         StringBuilder buf = new StringBuilder("Executing Maven: ");
         for (String arg : args)
-            buf.append(' ').append(arg);
+        {
+        	final String argPassword = "-Dpassword=" ;
+        	String filteredArg = arg ;
+        	// check if current arg is password arg. Then replace password by ***** 
+        	if ((arg.length() >= argPassword.length()) 
+        		&&  argPassword.equals(arg.substring(0, argPassword.length())) )
+        	{
+        		filteredArg=argPassword+"*********" ;
+        	}
+            buf.append(' ').append(filteredArg);
+        }
         return buf.toString();
     }
 
