--- C:/Users/ahalovanic/workspace/jenkinsci-promoted-builds-plugin-932a018/src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition - Copy.java	Thu Nov  8 09:17:12 2012
+++ C:/Users/ahalovanic/workspace/jenkinsci-promoted-builds-plugin-932a018/src/main/java/hudson/plugins/promoted_builds/conditions/ManualCondition.java	Thu Nov  1 17:31:28 2012
@@ -212,6 +212,8 @@
             for (ParameterValue value : getParameterValues()) {
                 value.buildEnvVars(build, env);
             }
+            env.put("PROMOTED_USERNAME", authenticationName);
+            env.put("PROMOTED_NAME", getUserName());
         }
     }
 
