Index: src/main/java/hudson/maven/MavenModule.java
===================================================================
--- src/main/java/hudson/maven/MavenModule.java (revision )
+++ src/main/java/hudson/maven/MavenModule.java (revision )
@@ -233,6 +233,7 @@
      * The path separator is normalized to '/'.
      */
     public String getRelativePath() {
+       if(relativePath == null) return "/";
         return relativePath;
     }
