Index: plugins/promoted-builds/src/main/java/hudson/plugins/promoted_builds/Promotion.java
===================================================================
--- plugins/promoted-builds/src/main/java/hudson/plugins/promoted_builds/Promotion.java	(revision 22642)
+++ plugins/promoted-builds/src/main/java/hudson/plugins/promoted_builds/Promotion.java	(working copy)
@@ -9,6 +9,7 @@
 import hudson.model.Node;
 import hudson.model.Result;
 import hudson.model.TaskListener;
+import hudson.model.TopLevelItem;
 import hudson.security.Permission;
 import hudson.security.PermissionGroup;
 import hudson.slaves.WorkspaceList;
@@ -82,7 +83,7 @@
 
         @Override
         protected Lease decideWorkspace(Node n, WorkspaceList wsl) throws InterruptedException, IOException {
-            return wsl.acquire(getTarget().getWorkspace(),true);
+            return wsl.acquire(n.getWorkspaceFor((TopLevelItem)getTarget().getProject()),true);
         }
 
         protected Result doRun(BuildListener listener) throws Exception {
