Index: src/main/java/hudson/plugins/bazaar/BazaarChangeLogParser.java
===================================================================
--- src/main/java/hudson/plugins/bazaar/BazaarChangeLogParser.java	(révision 27112)
+++ src/main/java/hudson/plugins/bazaar/BazaarChangeLogParser.java	(copie de travail)
@@ -75,13 +75,15 @@
                     break;
                 case 4:
                     if (!(ident == nident && s.startsWith("message:"))) {
-                        if (ident == nident && (s.startsWith("modified:") || s.startsWith("added:") || s.startsWith("removed:"))) {
+                        if (ident == nident && (s.startsWith("modified:") || s.startsWith("added:") || s.startsWith("removed:") || s.startsWith("renamed:"))) {
                             if (s.startsWith("modified")) {
                                 state = 5;
                             } else if (s.startsWith("added:")) {
                                 state = 6;
                             } else if (s.startsWith("removed:")) {
                                 state = 7;
+                            } else if (s.startsWith("renamed:")){
+                            	state = 8;
                             }
                             entry.setMsg(message.toString());
                             message.setLength(0);
@@ -91,56 +93,64 @@
                         }
                     }
                     break;
-                case 5:
+                case 5: // modified
                     if (s.startsWith("modified")) {
                         state = 5;
                     } else if (s.startsWith("added:")) {
                         state = 6;
                     } else if (s.startsWith("removed:")) {
                         state = 7;
+                    } else if (s.startsWith("renamed:")){
+                    	state = 8;
                     } else {
-                        int idx = s.indexOf(" => ");
-                        if (idx != -1) {
-                            s = s.substring(idx + 4);
-                        }
                         entry.getModifiedPaths().add(s);
 
                     }
 
                     break;
-                case 6:
+                case 6: // added
                     if (s.startsWith("modified")) {
                         state = 5;
                     } else if (s.startsWith("added:")) {
                         state = 6;
                     } else if (s.startsWith("removed:")) {
                         state = 7;
+                    } else if (s.startsWith("renamed:")){
+                    	state = 8;
                     } else {
-                        int idx = s.indexOf(" => ");
-                        if (idx != -1) {
-                            s = s.substring(idx + 4);
-                        }
                         entry.getAddedPaths().add(s);
                     }
 
                     break;
-                case 7:
+                case 7: // removed
                     if (s.startsWith("modified")) {
                         state = 5;
                     } else if (s.startsWith("added:")) {
                         state = 6;
                     } else if (s.startsWith("removed:")) {
                         state = 7;
+                    } else if (s.startsWith("renamed:")){
+                    	state = 8;
                     } else {
-                        int idx = s.indexOf(" => ");
-                        if (idx != -1) {
-                            s = s.substring(idx + 4);
-                        }
                         entry.getDeletedPaths().add(s);
                     }
 
                     break;
+                case 8: // renamed
+                    if (s.startsWith("modified")) {
+                        state = 5;
+                    } else if (s.startsWith("added:")) {
+                        state = 6;
+                    } else if (s.startsWith("removed:")) {
+                        state = 7;
+                    } else if (s.startsWith("renamed:")){
+                    	state = 8;
+                    } else {
+                        entry.getModifiedPaths().add(s);
+                    }
 
+                    break;
+
                 default: {
                     Logger logger = Logger.getLogger(BazaarChangeLogParser.class.getName());
                     logger.warning("Unknown parser state: " + state);
