### Eclipse Workspace Patch 1.0
#P qc
Index: src/main/resources/runTestSet.vbs
===================================================================
--- src/main/resources/runTestSet.vbs	(Revision 36999)
+++ src/main/resources/runTestSet.vbs	(Arbeitskopie)
@@ -159,7 +159,7 @@
     Else
         tdConnection.InitConnectionEx QCServerURL
         If tdConnection.Connected = False Then
-          errorMsg = "Cant connect to server"
+          errorMsg = "Can't connect to server"
         Else
             WScript.StdOut.WriteLine "Connected to server " & QCServerURL
             tdConnection.Login QCLogin, QCPass
@@ -224,71 +224,75 @@
         WScript.StdOut.WriteLine generateLine(50)
 
         Set Scheduler = targetTestSet.StartExecution("")
-        Set tsTestFactory = targetTestSet.TSTestFactory
-        Set tdFilter = tsTestFactory.Filter
-        tdFilter.Filter("TC_CYCLE_ID") = targetTestSet.ID
-        Set tList = tsTestFactory.NewList(tdFilter.Text)
-        WScript.StdOut.WriteBlankLines(2)
-        WScript.StdOut.WriteLine generateLine(50)
-        WScript.StdOut.WriteLine "| " & addBlankSpaces("Test Name", 15) &  " | " & addBlankSpaces("ID", 6) & " | " & addBlankSpaces("Host", 20) &  "|"
-        WScript.StdOut.WriteLine generateLine(50)
+	    If Scheduler Is Nothing Then
+	      errorMsg = "Could not instantiate test set scheduler"
+   	    Else
+          Set tsTestFactory = targetTestSet.TSTestFactory
+          Set tdFilter = tsTestFactory.Filter
+          tdFilter.Filter("TC_CYCLE_ID") = targetTestSet.ID
+          Set tList = tsTestFactory.NewList(tdFilter.Text)
+          WScript.StdOut.WriteBlankLines(2)
+          WScript.StdOut.WriteLine generateLine(50)
+          WScript.StdOut.WriteLine "| " & addBlankSpaces("Test Name", 15) &  " | " & addBlankSpaces("ID", 6) & " | " & addBlankSpaces("Host", 20) &  "|"
+          WScript.StdOut.WriteLine generateLine(50)
 
-        ReDim tests(tList.Count - 1)
+          ReDim tests(tList.Count - 1)
 
-        For Each test In tList
-          WScript.StdOut.WriteLine "| " & addBlankSpaces(test.Name, 15) &  " | " & addBlankSpaces(test.ID, 6) & " | " & addBlankSpaces(test.HostName, 20) &  "|"
-          WScript.StdOut.WriteLine generateLine(50)
-          Scheduler.RunOnHost(test.ID) = test.HostName
-        Next
+          For Each test In tList
+            WScript.StdOut.WriteLine "| " & addBlankSpaces(test.Name, 15) &  " | " & addBlankSpaces(test.ID, 6) & " | " & addBlankSpaces(test.HostName, 20) &  "|"
+            WScript.StdOut.WriteLine generateLine(50)
+            Scheduler.RunOnHost(test.ID) = test.HostName
+          Next
 
-        Scheduler.RunAllLocally = False
-        Scheduler.run
-        WScript.StdOut.WriteBlankLines(2)
-        WScript.StdOut.WriteLine "Running-Tests..."
-        WScript.StdOut.WriteBlankLines(2)
-        Set executionStatus = Scheduler.ExecutionStatus
+          Scheduler.RunAllLocally = False
+          Scheduler.run
+          WScript.StdOut.WriteBlankLines(2)
+          WScript.StdOut.WriteLine "Running-Tests..."
+          WScript.StdOut.WriteBlankLines(2)
+          Set executionStatus = Scheduler.ExecutionStatus
 
-        While ((tsExecutionFinished = False) And (iter < timeout))
-          iter = iter + 5
-          executionStatus.RefreshExecStatusInfo "all", True
-          tsExecutionFinished = executionStatus.Finished
-          WScript.Sleep( 5000 )
-        Wend
+          While ((tsExecutionFinished = False) And (iter < timeout))
+            iter = iter + 5
+            executionStatus.RefreshExecStatusInfo "all", True
+            tsExecutionFinished = executionStatus.Finished
+            WScript.Sleep( 5000 )
+          Wend
 
-        If iter < timeout Then
+          If iter < timeout Then
 
-          WScript.StdOut.WriteLine generateLine(50)
-          WScript.StdOut.WriteLine "| " & addBlankSpaces("Test", 22) &  " | " & addBlankSpaces("Result", 22) & "|"
-          WScript.StdOut.WriteLine generateLine(50)
+            WScript.StdOut.WriteLine generateLine(50)
+            WScript.StdOut.WriteLine "| " & addBlankSpaces("Test", 22) &  " | " & addBlankSpaces("Result", 22) & "|"
+            WScript.StdOut.WriteLine generateLine(50)
 
-          For i = 1 To executionStatus.Count
-            Set qTest = New QCTest
-            Set testExecStatusObj = executionStatus.Item(i)
+            For i = 1 To executionStatus.Count
+              Set qTest = New QCTest
+              Set testExecStatusObj = executionStatus.Item(i)
 
-            Set currentTest = targetTestSet.TSTestFactory.Item(testExecStatusObj.TSTestId)
+              Set currentTest = targetTestSet.TSTestFactory.Item(testExecStatusObj.TSTestId)
 
-            qTest.Name = currentTest.Name
-            qTest.Duration = currentTest.LastRun.Field("RN_DURATION")
+              qTest.Name = currentTest.Name
+              qTest.Duration = currentTest.LastRun.Field("RN_DURATION")
 
-            If Not (currentTest.LastRun.Status = "Passed") Then
-              Set qFailure = New QCFailure
-              qFailure.Desc = testExecStatusObj.Message
-              qFailure.Name = currentTest.LastRun.Status
-              Set qTest.Failure = qFailure
-            Else
-              Set qTest.Failure = Nothing
-            End If
+              If Not (currentTest.LastRun.Status = "Passed") Then
+                Set qFailure = New QCFailure
+                qFailure.Desc = testExecStatusObj.Message
+                qFailure.Name = currentTest.LastRun.Status
+                Set qTest.Failure = qFailure
+              Else
+                Set qTest.Failure = Nothing
+              End If
 
-            Set tests(i - 1) = qTest
-              WScript.StdOut.WriteLine "| " & addBlankSpaces(currentTest.Name, 22) &  " | " & addBlankSpaces(currentTest.LastRun.Status, 22) & "|"
-              WScript.StdOut.WriteLine generateLine(50)
-          Next
+              Set tests(i - 1) = qTest
+                WScript.StdOut.WriteLine "| " & addBlankSpaces(currentTest.Name, 22) &  " | " & addBlankSpaces(currentTest.LastRun.Status, 22) & "|"
+                WScript.StdOut.WriteLine generateLine(50)
+            Next
 
-          WScript.StdOut.WriteBlankLines(2)
-          WScript.StdOut.WriteLine "Scheduler finished around " & CStr(Now)
-        Else
-          errorMsg = "Timed out"
-        End If
+            WScript.StdOut.WriteBlankLines(2)
+            WScript.StdOut.WriteLine "Scheduler finished around " & CStr(Now)
+          Else
+            errorMsg = "Timed out"
+          End If
+	    End If
       End If
     End If
   End Sub
