Index: src/main/java/hudson/plugins/perforce/HudsonPipedOutputStream.java
===================================================================
--- src/main/java/hudson/plugins/perforce/HudsonPipedOutputStream.java	(revision 36969)
+++ src/main/java/hudson/plugins/perforce/HudsonPipedOutputStream.java	(working copy)
@@ -4,6 +4,7 @@
 import java.io.PipedOutputStream;
 
 import hudson.Proc;
+import hudson.remoting.FastPipedOutputStream;
 
 /**
  * Extended class of PipedOutputStream, used specifically for passing into
@@ -24,7 +25,7 @@
  *
  * @author Victor Szoltysek
  */
-public class HudsonPipedOutputStream extends PipedOutputStream {
+public class HudsonPipedOutputStream extends FastPipedOutputStream {
 
 	// Close stream, when hudson process finishes.
 	public void closeOnProcess(final Proc process) {
Index: src/main/java/hudson/plugins/perforce/HudsonP4Executor.java
===================================================================
--- src/main/java/hudson/plugins/perforce/HudsonP4Executor.java	(revision 36969)
+++ src/main/java/hudson/plugins/perforce/HudsonP4Executor.java	(working copy)
@@ -10,6 +10,8 @@
 import hudson.Launcher;
 import hudson.Proc;
 import hudson.model.Hudson;
+import hudson.remoting.FastPipedInputStream;
+import hudson.remoting.FastPipedOutputStream;
 import hudson.util.StreamTaskListener;
 
 /**
@@ -63,12 +65,12 @@
 
             // hudsonOut->p4in->reader
             HudsonPipedOutputStream hudsonOut = new HudsonPipedOutputStream();
-            PipedInputStream p4in = new PipedInputStream(hudsonOut);
+            FastPipedInputStream p4in = new FastPipedInputStream(hudsonOut);
             reader = new BufferedReader(new InputStreamReader(p4in));
 
             // hudsonIn<-p4Out<-writer
-            PipedInputStream hudsonIn = new PipedInputStream();
-            PipedOutputStream p4out = new PipedOutputStream(hudsonIn);
+            FastPipedInputStream hudsonIn = new FastPipedInputStream();
+            FastPipedOutputStream p4out = new FastPipedOutputStream(hudsonIn);
             writer = new BufferedWriter(new OutputStreamWriter(p4out));
 
             Proc process = hudsonLauncher.launch().cmds(cmd).envs(env).stdin(hudsonIn).stdout(hudsonOut).pwd(filePath).start();
