diff --git a/opensuse/SPECS/jenkins.spec b/opensuse/SPECS/jenkins.spec
index d7314fc..44a7a00 100644
--- a/opensuse/SPECS/jenkins.spec
+++ b/opensuse/SPECS/jenkins.spec
@@ -88,8 +88,6 @@ rm -rf "%{buildroot}"
        -d "%{workdir}" jenkins &>/dev/null || :
 
 %post
-[ $1 -eq 1 ] && /sbin/chkconfig --add jenkins
-
 # If we have an old hudson install, rename it to jenkins
 if test -d /var/lib/hudson; then
     # leave a marker to indicate this came from Hudson.

