diff --git a/src/main/java/hudson/plugins/analysis/core/HealthAwareReporter.java b/src/main/java/hudson/plugins/analysis/core/HealthAwareReporter.java
index f8048df..19a2a5d 100644
--- a/src/main/java/hudson/plugins/analysis/core/HealthAwareReporter.java
+++ b/src/main/java/hudson/plugins/analysis/core/HealthAwareReporter.java
@@ -246,8 +246,16 @@
      *
      * @return <code>true</code> if the previous build should always be used.
      */
-    public boolean getUsePreviousBuildAsStable() {
+    public boolean getUsePreviousBuildAsReference() {
         return usePreviousBuildAsReference;
+    }
+
+    /**
+     * @deprecated mistyped method name from v1.72 - see {@link #getUsePreviousBuildAsReference()}
+     */
+    @Deprecated
+    public boolean getUsePreviousBuildAsStable() {
+        return getUsePreviousBuildAsReference();
     }
 
     /**
@@ -255,8 +263,16 @@
      *
      * @return <code>true</code> if the previous build should always be used.
      */
+    public boolean usePreviousBuildAsReference() {
+        return getUsePreviousBuildAsReference();
+    }
+
+    /**
+     * @deprecated mistyped method name from v1.72 - see {@link #usePreviousBuildAsReference()}
+     */
+    @Deprecated
     public boolean usePreviousBuildAsStable() {
-        return getUsePreviousBuildAsStable();
+        return usePreviousBuildAsReference();
     }
 
     /**