### Eclipse Workspace Patch 1.0
#P synergy
Index: src/main/java/hudson/plugins/synergy/SynergySCM.java
===================================================================
--- src/main/java/hudson/plugins/synergy/SynergySCM.java	(revision 21039)
+++ src/main/java/hudson/plugins/synergy/SynergySCM.java	(working copy)
@@ -455,7 +455,7 @@
 
 	private CheckoutResult checkoutDynamicProject(FilePath path, File changeLogFile, String projectName) throws IOException, InterruptedException, SynergyException {
 		// Configure workarea.
-		setAbsoluteWorkarea(path);
+		setAbsoluteWorkarea(path, projectName);
 
 		// Update members.
 		UpdateCommand updateCommand = new UpdateCommand(projectName, isReplaceSubprojects());
@@ -619,12 +619,12 @@
 	 * @throws InterruptedException
 	 * @throws SynergyException
 	 */
-	private void setAbsoluteWorkarea(FilePath path) throws IOException, InterruptedException, SynergyException {
+	private void setAbsoluteWorkarea(FilePath path, String projectName) throws IOException, InterruptedException, SynergyException {
 		// Check main project.
-		configureWorkarea(project, false, path);
+		configureWorkarea(projectName, false, path);
 
 		// Get subproject.
-		SubProjectQueryCommand command = new SubProjectQueryCommand(project);
+		SubProjectQueryCommand command = new SubProjectQueryCommand(projectName);
 		try {
 			commands.executeSynergyCommand(path, command);
 		} catch (SynergyException e) {
