Index: src/main/java/org/jvnet/hudson/plugins/backup/utils/BackupTask.java
===================================================================
--- src/main/java/org/jvnet/hudson/plugins/backup/utils/BackupTask.java	(revision 23777)
+++ src/main/java/org/jvnet/hudson/plugins/backup/utils/BackupTask.java	(working copy)
@@ -114,15 +114,6 @@
     }
 
     private void waitNoJobsInQueue() {
-        logger
-                .info("Setting hudson in shutdown mode to avoid files corruptions.");
-        try {
-            Hudson.getInstance().doQuietDown();
-        } catch (Exception e) {
-            logger.error("Erreur putting hudson in shutdown mode.");
-            e.printStackTrace(logger.getWriter());
-        }
-
         logger.info("Waiting all jobs end...");
 
         Computer computer[] = Hudson.getInstance().getComputers();
@@ -136,6 +127,10 @@
             }
             logger.info("Number of running jobs detected : " + running);
 
+            if (running == 0) {
+                break;
+            }
+
             try {
                 Thread.sleep(DELAY);
             } catch (InterruptedException e) {
@@ -143,6 +138,16 @@
         }
 
         logger.info("All jobs finished.");
+
+        logger
+                .info("Setting hudson in shutdown mode to avoid files corruptions.");
+        try {
+            Hudson.getInstance().doQuietDown();
+        } catch (Exception e) {
+            logger.error("Erreur putting hudson in shutdown mode.");
+            e.printStackTrace(logger.getWriter());
+        }
+
     }
 
     private void cancelNoJobs() {
