--- AccurevTransaction.java	2011-10-18 12:37:10.035769200 -0400
+++ del.java	2011-10-18 12:38:57.654608100 -0400
@@ -64,7 +64,7 @@
     }
 
     public String getMsg() {
-        return msg;
+        return (null == msg ? "" : msg);
     }
 
     public void setMsg(String msg) {
@@ -137,7 +137,7 @@
                 FIELD_SEPARATOR + //
                 "action" + EQ + action + //
                 FIELD_SEPARATOR + //
-                "msg" + EQ + (msg == null ? "" : msg) + //
+                "msg" + EQ + getMsg() + //
                 ']';
     }
 }
