Index: src/main/java/hudson/plugins/logparser/LogParserBuildStep.java
===================================================================
--- src/main/java/hudson/plugins/logparser/LogParserBuildStep.java	(revision 34008)
+++ src/main/java/hudson/plugins/logparser/LogParserBuildStep.java	(working copy)
@@ -9,6 +9,8 @@
 import hudson.tasks.BuildStep;
 import hudson.tasks.BuildStepMonitor;
 import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
 
 public class LogParserBuildStep implements BuildStep  {
 
@@ -30,5 +32,10 @@
 	public Action getProjectAction(final AbstractProject<?, ?> project) {
 		return null;
 	}
+
+  public Collection<? extends Action> getProjectActions(
+      AbstractProject<?, ?> project) {
+    return Collections.emptyList();
+  }
 }
 
Index: src/main/java/hudson/plugins/logparser/LogParserParser.java
===================================================================
--- src/main/java/hudson/plugins/logparser/LogParserParser.java	(revision 34008)
+++ src/main/java/hudson/plugins/logparser/LogParserParser.java	(working copy)
@@ -1,6 +1,7 @@
 package hudson.plugins.logparser;
 
 
+import hudson.console.ConsoleNote;
 import hudson.model.AbstractBuild;
 import java.io.*;
 import java.util.HashMap;
@@ -169,6 +170,7 @@
 		if (status.equals(LogParserConsts.START)) {
 			effectiveStatus = LogParserConsts.INFO;
 		}
+		parsedLine = ConsoleNote.removeNotes(parsedLine);
 		parsedLine = parsedLine.replaceAll("<", "&lt;"); // Allows < to be seen in log which is html
 		parsedLine = parsedLine.replaceAll(">", "&gt;"); // Allows > to be seen in log which is html
 		if (effectiveStatus != null && !effectiveStatus.equals(LogParserConsts.NONE)) {
Index: pom.xml
===================================================================
--- pom.xml	(revision 34008)
+++ pom.xml	(working copy)
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.jvnet.hudson.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>1.340</version>
+    <version>1.350</version>
     <relativePath>../pom.xml</relativePath>
   </parent>
   
