diff --git a/src/main/java/hudson/plugins/performance/PerformanceBuildAction.java b/src/main/java/hudson/plugins/performance/PerformanceBuildAction.java
index 6d584fb..79e64e1 100644
--- a/src/main/java/hudson/plugins/performance/PerformanceBuildAction.java
+++ b/src/main/java/hudson/plugins/performance/PerformanceBuildAction.java
@@ -34,10 +34,10 @@ public class PerformanceBuildAction implements Action, StaplerProxy {
     this.parsers = parsers;
   }
 
-  public PerformanceReportParser getParserById(String id) {
+  public PerformanceReportParser getParserByDisplayName(String displayName) {
     if (parsers != null)
       for (PerformanceReportParser parser : parsers)
-        if (parser.getDescriptor().getId().equals(id))
+        if (parser.getDescriptor().getDisplayName().equals(displayName))
           return parser;
     return null;
   }
diff --git a/src/main/java/hudson/plugins/performance/PerformanceProjectAction.java b/src/main/java/hudson/plugins/performance/PerformanceProjectAction.java
index 688c591..ab35e0d 100644
--- a/src/main/java/hudson/plugins/performance/PerformanceProjectAction.java
+++ b/src/main/java/hudson/plugins/performance/PerformanceProjectAction.java
@@ -336,20 +336,22 @@ public final class PerformanceProjectAction implements Action {
 
   public List<String> getPerformanceReportList() {
     this.performanceReportList = new ArrayList<String>(0);
-    if (this.project != null
-        && this.project.getSomeBuildWithWorkspace() != null) {
-      File file = new File(
-          this.project.getSomeBuildWithWorkspace().getRootDir(),
-          PerformanceReportMap.getPerformanceReportDirRelativePath());
-      if (file.exists()) {
-        for (File performanceReportFile : file.listFiles()) {
-          this.performanceReportList.add(performanceReportFile.getName());
-        }
+    if (null == this.project) { return performanceReportList; }
+    if (null == this.project.getSomeBuildWithWorkspace()) { return performanceReportList; }
+    File file = new File(
+        this.project.getSomeBuildWithWorkspace().getRootDir(),
+        PerformanceReportMap.getPerformanceReportDirRelativePath());
+    if (!file.isDirectory()) { return performanceReportList; }
+    
+    for (File entry : file.listFiles()) {
+      if (entry.isDirectory()) {
+    	for (File e : entry.listFiles()) { this.performanceReportList.add(e.getName()); }
+      } else {
+    	this.performanceReportList.add(entry.getName());
       }
     }
-    if (this.performanceReportList != null) {
-      Collections.sort(performanceReportList);
-    }
+    
+    Collections.sort(performanceReportList);
     return this.performanceReportList;
   }
 
diff --git a/src/main/java/hudson/plugins/performance/PerformancePublisher.java b/src/main/java/hudson/plugins/performance/PerformancePublisher.java
index e577877..55f6e3c 100644
--- a/src/main/java/hudson/plugins/performance/PerformancePublisher.java
+++ b/src/main/java/hudson/plugins/performance/PerformancePublisher.java
@@ -74,10 +74,10 @@ public class PerformancePublisher extends Recorder {
   }
 
   public static File getPerformanceReport(AbstractBuild<?, ?> build,
-      String performanceReportName) {
+      String parserDisplayName, String performanceReportName) {
     return new File(
         build.getRootDir(),
-        PerformanceReportMap.getPerformanceReportFileRelativePath(getPerformanceReportBuildFileName(performanceReportName)));
+        PerformanceReportMap.getPerformanceReportFileRelativePath(parserDisplayName, getPerformanceReportBuildFileName(performanceReportName)));
   }
 
   @Override
@@ -195,7 +195,7 @@ public class PerformancePublisher extends Recorder {
         return true;
       }
 
-      List<File> localReports = copyReportsToMaster(build, logger, files);
+      List<File> localReports = copyReportsToMaster(build, logger, files, parser.getDescriptor().getDisplayName());
       Collection<PerformanceReport> parsedReports = parser.parse(build,
           localReports, listener);
 
@@ -220,11 +220,11 @@ public class PerformancePublisher extends Recorder {
   }
 
   private List<File> copyReportsToMaster(AbstractBuild<?, ?> build,
-      PrintStream logger, List<FilePath> files) throws IOException,
+      PrintStream logger, List<FilePath> files, String parserDisplayName) throws IOException,
       InterruptedException {
     List<File> localReports = new ArrayList<File>();
     for (FilePath src : files) {
-      final File localReport = getPerformanceReport(build, src.getName());
+      final File localReport = getPerformanceReport(build, parserDisplayName, src.getName());
       if (src.isDirectory()) {
         logger.println("Performance: File '" + src.getName()
             + "' is a directory, not a Performance Report");
diff --git a/src/main/java/hudson/plugins/performance/PerformanceReportMap.java b/src/main/java/hudson/plugins/performance/PerformanceReportMap.java
index 91d106b..3b65769 100644
--- a/src/main/java/hudson/plugins/performance/PerformanceReportMap.java
+++ b/src/main/java/hudson/plugins/performance/PerformanceReportMap.java
@@ -72,7 +72,7 @@ public class PerformanceReportMap implements ModelObject {
     // this may fail, if the build itself failed, we need to recover gracefully
     if (dirs != null) {
       for (File dir : dirs) {
-        PerformanceReportParser p = buildAction.getParserById(dir.getName());
+        PerformanceReportParser p = buildAction.getParserByDisplayName(dir.getName());
         if (p != null) {
           addAll(p.parse(getBuild(), Arrays.asList(dir.listFiles()), listener));
         }
@@ -162,19 +162,19 @@ public class PerformanceReportMap implements ModelObject {
   }
 
   public static String getPerformanceReportFileRelativePath(
-      String reportFileName) {
-    return getRelativePath(reportFileName);
+      String parserDisplayName, String reportFileName) {
+    return getRelativePath(parserDisplayName, reportFileName);
   }
 
   public static String getPerformanceReportDirRelativePath() {
-    return getRelativePath(null);
+    return getRelativePath();
   }
 
-  private static String getRelativePath(String reportFileName) {
+  private static String getRelativePath(String... suffixes) {
     StringBuilder sb = new StringBuilder(100);
     sb.append(PERFORMANCE_REPORTS_DIRECTORY);
-    if (reportFileName != null) {
-      sb.append("/").append(reportFileName);
+    for (String suffix : suffixes) {
+    	sb.append(File.separator).append(suffix);
     }
     return sb.toString();
   }
