Index: src/main/resources/hudson/plugins/ec2/AmazonEC2Cloud/config-entries.jelly
===================================================================
--- src/main/resources/hudson/plugins/ec2/AmazonEC2Cloud/config-entries.jelly	(revision 29921)
+++ src/main/resources/hudson/plugins/ec2/AmazonEC2Cloud/config-entries.jelly	(working copy)
@@ -16,6 +16,6 @@
       <f:textbox />
     </f:entry>
   </f:advanced>
-  <f:validateButton title="${%Generate Key}" progress="${%Generate...}" method="generateKey" with="secretKey,accessId" />
+  <f:validateButton title="${%Generate Key}" progress="${%Generate...}" method="generateKey" with="region,secretKey,accessId" />
   <f:validateButton title="${%Test Connection}" progress="${%Testing...}" method="testConnection" with="region,secretKey,accessId,privateKey" />
 </j:jelly>
