Index: TypeDescriptor.java
===================================================================
--- TypeDescriptor.java	(revision 21388)
+++ TypeDescriptor.java	(working copy)
@@ -10,6 +10,7 @@
 import hudson.plugins.violations.types.gendarme.GendarmeDescriptor;
 import hudson.plugins.violations.types.cpd.CPDDescriptor;
 import hudson.plugins.violations.types.pylint.PyLintDescriptor;
+import hudson.plugins.violations.types.javascriptlint.JavascriptLintDescriptor;
 import hudson.plugins.violations.types.simian.SimianDescriptor;
 import hudson.plugins.violations.types.stylecop.StyleCopDescriptor;
 
@@ -81,6 +82,7 @@
         addDescriptor(CPDDescriptor.DESCRIPTOR);
         addDescriptor(CheckstyleDescriptor.DESCRIPTOR);
         addDescriptor(PyLintDescriptor.DESCRIPTOR);
+        addDescriptor(JavascriptLintDescriptor.DESCRIPTOR);
         addDescriptor(FxCopDescriptor.DESCRIPTOR);
         addDescriptor(SimianDescriptor.DESCRIPTOR);
         addDescriptor(StyleCopDescriptor.DESCRIPTOR);
